pkg/broker/filter/filter_handler.go

```diff
@@ -206,34 +222,60 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 
 	h.reportArrivalTime(event, reportArgs)
 
-	h.send(ctx, writer, request.Header, subscriberURI.String(), reportArgs, event, ttl)
+	h.send(ctx, writer, request.Header, subscriberURI.URL(), reportArgs, event, ttl)
 }
 
-func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target string, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
+func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target *url.URL, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
 	// send the event to trigger's subscriber
-	response, err := h.sendEvent(ctx, headers, target, event, reportArgs)
-	if err != nil {
-		h.logger.Error("failed to send event", zap.Error(err))
-		writer.WriteHeader(http.StatusInternalServerError)
-		_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+	response, responseErr := h.sendEvent(ctx, headers, target, event, reportArgs)
+
+	if responseErr.err != nil {
+		h.logger.Error("failed to send event", zap.Error(responseErr.err))
+		// If error is not because of the response, it should respond with http.StatusInternalServerError
+		if responseErr.ResponseCode == NoResponse {
+
+			writer.WriteHeader(http.StatusInternalServerError)
+			_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+			return
+		}
+
+		writer.WriteHeader(responseErr.ResponseCode)
+		// Read Response body to responseErr
+		errExtensionInfo := broker.ErrExtensionInfo{
+			ErrDestination:  target,
+			ErrResponseBody: responseErr.ResponseBody,
+		}
+		errExtensionBytes, msErr := json.Marshal(errExtensionInfo)
+		if msErr != nil {
+			h.logger.Error("failed to marshal errExtensionInfo", zap.Error(msErr))
+			return
+		}
+		_, _ = writer.Write(errExtensionBytes)
```

Comment:
We're missing the call to:
```
		proxyHeaders(resp.Header, writer) // Proxy original Response Headers for downstream use
```

Reply:
So, if we include the entire response in ErrHandler, we can read and proxy headers